keps/sig-scheduling/5007-device-attach-before-pod-scheduled/README.md

```diff
@@ -0,0 +1,1111 @@
+
+![proposal](proposal.jpg)
+
+#### BasicDevice Enhancements
+
+To indicate that the device is a Fabric device or other device that requires some preparation (e.g. attachment), some fields are added to the `Basic` within `Device`.
+These fields will be used by the controller that exposes the `ResourceSlice` to notify whether the device is a fabric device.
+
+```go
+// BasicDevice represents a basic device instance.
+type BasicDevice struct {
+    // BindingConditions defines the conditions for proceeding with binding. All listed conditions must be True to proceed with binding. 
+    //
+    // +optional
+    BindingConditions []string
+
+    // BindingFailureConditions defines the conditions for binding failure.
+    // If true, a binding failure occurred.
+    //
+    // +optional
+    BindingFailureConditions []string
+
+    // UsageRestrictedToNode indicates if the usage of an allocation involving this device
+    // has to be limited to exactly the node that was chosen when allocating the claim.
+    //
+    // +optional
+    UsageRestrictedToNode bool
+
+    // BindingTimeout indicates the prepare timeout period.
+    // If the timeout period is exceeded before all BindingConditions reach a True state, the scheduler clears the allocation in the ResourceClaim and reschedules the Pod.
+    //
+    // +optional
+    BindingTimeout *metav1.Duration
+}
+
+const (
+    BindingConditionsMaxSize    = 4
+    BindingFailureConditionsMaxSize = 4
+)
+```
+
+#### AllocatedDeviceStatus Enhancements
+
+The `BindingConditions` and `BindingFailureConditions` fields within `AllocatedDeviceStatus` are used to indicate the status of the device attachment.
+These fields will contain a list of conditions, each representing a specific state or event related to the device.
+
+For this feature, following fields are added:
+
+```go
+// AllocatedDeviceStatus contains the status of an allocated device, if the
+// driver chooses to report it. This may include driver-specific information.
+type AllocatedDeviceStatus struct {
+    ...
+    // BindingConditions defines the conditions for proceeding with binding. All listed conditions must be True to proceed with binding. 
+    //
+    // +optional
+    BindingConditions []string
+
+    // BindingFailureConditions defines the conditions for binding failure.
+    // If any is True, a binding failure occurred.
+    //
+    // +optional
+    BindingFailureConditions []string
+
+    // UsageRestrictedToNode indicates if the usage of an allocation involving this device
+    // has to be limited to exactly the node that was chosen when allocating the claim.
+    //
+    // +optional
+    UsageRestrictedToNode bool
+
+    // BindingTimeout indicates the prepare timeout period.
+    // If the timeout period is exceeded before all BindingConditions reach a True state, the scheduler clears the allocation in the ResourceClaim and reschedules the Pod.
+    //
+    // +optional
```

Comment:
```suggestion
    //  The default timeout if not set is 10 minutes.
    //
    // +optional
```

Reply:
Thank you for your review. I have reflected all of your suggestions and comments.